src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts

```diff
@@ -713,3 +713,26 @@ g.test('multisample_render_target_formats_support_resolve')
       colorAttachments: [colorAttachment],
     });
   });
+
+g.test('occlusion_query_set_type')
+  .desc('Test that occlusion query set only works if the type is occlusion')
+  .paramsSimple([
+    { querySetState: 'valid' as const, queryType: 'occlusion', _success: true },
+    { querySetState: 'invalid' as const, queryType: 'timestamp', _success: false },
+  ])
```

Comment:
nit: `as const` avoids needing `as GPUQueryType` below (it causes TypeScript to constrain the type of `t.params.queryType` more tightly)
```suggestion
  ] as const)
```